lib/net.c

```diff
@@ -43,16 +44,45 @@
 static struct net_config config;
 
 /*
- * XXX The secret key of the RSS hash must be random
- * in order to avoid hackers to know it.
+ * The secret key of the RSS hash (RSK) must be random in order
+ * to prevent hackers from knowing it.
  */
-uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
-	0x6d, 0x5a, 0x56, 0xda, 0x25, 0x5b, 0x0e, 0xc2,
-	0x41, 0x67, 0x25, 0x3d, 0x43, 0xa3, 0x8f, 0xb0,
-	0xd0, 0xca, 0x2b, 0xcb, 0xae, 0x7b, 0x30, 0xb4,
-	0x77, 0xcb, 0x2d, 0xa3, 0x80, 0x30, 0xf2, 0x0c,
-	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
-};
+uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN];
+
+/* Function to randomize the default RSS key. */
+static void
+randomize_rss_key(void)
+{
+	uint8_t i;
+	uint8_t final_set_count = 0;
+	ssize_t number_of_bytes;
+	uint8_t min_num_set_bits, max_num_set_bits;
+
+	number_of_bytes = getrandom(default_rss_key, sizeof(default_rss_key), 0);
+	while (number_of_bytes != sizeof(default_rss_key)) {
+		number_of_bytes = getrandom(default_rss_key, sizeof(default_rss_key), 0);
+	}
+
+	/* 
+	 * To validate if the key generated is reasonable, the
+	 * number of bits set to 1 in the key must be greater than 
+	 * 10% and less than 90% of the total bits in the key.
+	 * min_num_set_bits and max_num_set_bits represent the lower 
+	 * and upper bound for the key.
+	 */
+	min_num_set_bits = 32;
+	max_num_set_bits = 288;
+
+	for (i = 0; i < sizeof(default_rss_key); i++) {
+		uint8_t temp_count = __builtin_popcount(default_rss_key[i]);
```

Comment:
I think you can directly use ``final_set_count += __builtin_popcount(default_rss_key[i]);``, and remove the brackets, since the ``for`` loop will only have one line of code in its block if the suggestion is used.